federation/pkg/federation-controller/service/service_controller.go

```diff
@@ -165,11 +165,11 @@ func NewServiceController(client fedclientset.Interface) *ServiceController {
 		},
 		&util.ClusterLifecycleHandlerFuncs{
 			ClusterAvailable: func(cluster *fedv1.Cluster) {
-				// When new cluster becomes available process all the services again.
+				// When new cluster becomes available, process all the services again.
 				sc.clusterDeliverer.DeliverAfter(allServicesKey, nil, sc.clusterAvailableDelay)
 			},
 			ClusterUnavailable: func(cluster *fedv1.Cluster, _ []interface{}) {
-				// When cluster becomes unavailable process all the services again.
+				// When cluster is unregistered, process all the services again.
```

Comment:
cluster unregister event is a subset of of cluster unavailable event, and i was trying to handle unregister within unavailable event. To be precise, we should add a new event to federated_informer to be called upon cluster unregisteration. i will update the code for this.